src/main/java/bio/terra/cli/app/LocalProcessCommandRunner.java

```diff
@@ -35,20 +35,39 @@ protected String wrapCommandInSetupCleanup(List<String> command) {
     List<String> bashCommands = new ArrayList<>();
     boolean isGcpWorkspace =
         CloudPlatform.GCP.equals(Context.requireWorkspace().getCloudPlatform());
+
+    String cmdPre = "";
+    String cmdPost = "";
+
     if (isGcpWorkspace) {
-      bashCommands.add("echo 'Setting the gcloud project to the workspace project'");
-      bashCommands.add("TERRA_PREV_GCLOUD_PROJECT=$(gcloud config get-value project)");
-      bashCommands.add(
-          "gcloud config set project " + Context.requireWorkspace().getRequiredGoogleProjectId());
+      cmdPre = String.format("""
+        TERRA_GCLOUD_PROJECT='%s'
+        """, Context.requireWorkspace().getRequiredGoogleProjectId()) +
+        """
+        TERRA_PREV_GCLOUD_PROJECT="$(gcloud config get-value project)"
+        if [[ "${TERRA_PREV_GCLOUD_PROJECT}" != "${TERRA_GCLOUD_PROJECT}" ]]; then
+          echo "Setting the gcloud project to the workspace project: ${TERRA_GCLOUD_PROJECT}"
+          gcloud config set project "${TERRA_GCLOUD_PROJECT}"
```

Comment:
this looks fine.  but instead of modifying the config, can we not set this project via environment variable or gcloud flag?

i dislike how we mess with the user's config.

Reply:
@cbookg this is a great point. Have never liked this but had assumed this had been previously well researched. It turns out that we have a few choices as noted here:

https://cloud.google.com/sdk/docs/configurations

A broad solution we could look at is to set up and manage one or more terra-specific `gcloud` configuration directories. But in the short-term, we can simply use the environment variable `CLOUDSDK_CORE_PROJECT`. I just did some testing and validated that `gcloud` and `gsutil` respect this value.

I think this would be a pretty robust solution (and obviously better than what the cli current does). I was a bit concerned about the case where a tool that someone is calling in there `terra` subcommand picks up the project *from gcloud*, but this case looks consistently handled:

```
$ CLOUDSDK_CORE_PROJECT=terra-luminous-olive-1234 gcloud config list
[core]
account = <snip>
disable_usage_reporting = True
project = terra-luminous-olive-1234
```

```
$ CLOUDSDK_CORE_PROJECT=terra-luminous-olive-1234 gcloud config get project
terra-luminous-olive-1234
```
```
$ CLOUDSDK_CORE_PROJECT=terra-vdevel-windy-radish-1234 gcloud config get project
terra-vdevel-windy-radish-1234
```

Will change to using the environment variable.

Thanks for the prod!

Reply:
amazing, so much better!